src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
There is a window of time where `_total` has less than it did before. Although this window of time is short, it could be confusing and can be avoided by incrementing `_total` by the different (calculated and stored in a local).

Also, use `TryGetValue` to avoid the additional lookup.

Also, this method is setter is NOT thread safe. This is fine because we have a single thread loading aux data, right? Please leave a loud comment stating this.